File: 릿코드/167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
const twoSum = (numbers, target) => {
    const map = {};
    for(let i=0, len=numbers.length; i<len; i++) {
        const before = target-numbers[i];
        if(before in map) return [map[before]+1, i+1];
        map[numbers[i]] = i;
    }
};

File: 릿코드/167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        Map<Integer, Integer> map = new TreeMap<>();
        for(int i=0, len=numbers.length; i<len; i++) {
            if(map.keySet().contains(target-numbers[i])) return new int[]{map.get(target-numbers[i])+1, i+1};
            map.put(numbers[i], i);
        }
        return null;
    }
}